### ISSUE TITLE
Black video on A-side owr to owr 

### ISSUE THREAD

**Issue by stefanalund** [EricssonResearch/openwebrtc#168]
Only getting black *remote* video frames on A-side when running OWR -> OWR, e.g. Safari+daemon -> Bowser:

```
0:00:52.768842000  1698 0x16472f80 WARN           intervideosrc /Users/stefan/Developer/github/openwebrtc/scripts/dependencies/gst-plugins-bad/gst/inter/gstintervideosrc.c:427:gst_inter_video_src_create:<source-2> Creating black frame
```

**Comment by superdump** [comment]
Every time?

**Comment by stefanalund** [comment]
Yes. Using solid wifi at home. OWR to Chrome works every time though, even if I get the occasional black frame on Chrome (remote).

**Comment by superdump** [comment]
I just tried with the daemon and Safari with two tabs and I get black on both sides. Nice...

**Comment by stefanalund** [comment]
Tried rolling back to 1fe9cd79999ce3a9e4017953297548029aff9b40 (before RTX in JavaScript) and I see the same problem.

**Comment by superdump** [comment]
The black on both sides was with a cerbero build from yesterday on OS X. Using our build scripts, I get black on the initiator side, the same as Stefan reported.

**Comment by stefanalund** [comment]
Rolled back to https://github.com/EricssonResearch/openwebrtc/commit/47200c674466c6dc5126557e6462f13b7db6c3bb and rebuilt everything, still get the same problem.

**Comment by sdroege** [comment]
I have the same problem with test-send-receive apparently (or something else wrong there), will look into this.

**Comment by stefanalund** [comment]
Much appreciated, this is not so nice :-)

**Comment by sdroege** [comment]
This is fixed now by this commit: http://cgit.freedesktop.org/gstreamer/gst-plugins-good/commit/?id=b79eff7f9bfcf1b3b0360d346cc3b3483da1361e

However it still takes some time because we are losing the first keyframe again, and first have to re-request it. Not too long ago we did not lose the first keyframe.

**Comment by sdroege** [comment]
This is probably happening because of all the following debug outputs in the beginning:
0:00:00.699166552 29048      0x190ed90 WARN                 srtpdec gstsrtpdec.c:1025:gst_srtp_dec_chain:<srtp-decoder> Invalid buffer, dropping

Not sure why yet.

**Comment by stefanalund** [comment]
Fantastic, do we have to move to a newer version of GStreamer before this is propagated to owr?

**Comment by sdroege** [comment]
Yes, I'm also checking why we lose the key frame now. That's not good and delays getting the remote view here by 8 seconds

**Comment by stefanalund** [comment]
A lot better than getting no video at all :)

**Comment by superdump** [comment]
Excellent. I'm working on a build script update that I'll send up soon.

**Comment by ijsf** [comment]
Can confirm this (or a similar) issue, just tested with current openwebrtc built with Cerbero on an iPad Mini 1.

Client A (OpenWebRTC/Bowser, iOS 8.1.3): local stream works, remote stream (B) black
Client B (Chrome, OS X): local stream works, remote stream (A) black

**Comment by ijsf** [comment]
I'm suspecting the stream is coming in either RGBA or BGRA into gst_ercolorspace_transform_frame in the ercolorspace plugin. This function is not set up to handle RGBA or BGRA despite the capabilities indicating that it can.

**Comment by stefanalund** [comment]
Will this be fixed by moving to a newer version of GStreamer?

**Comment by superdump** [comment]
My understanding is newer GStreamer + some form of colour space conversion fix.

**Comment by sdroege** [comment]
The colorspace conversion workaround is there already, using videoconvert now. So only needs newer GStreamer, and for the initial black frames it needs the fix I'm currently working on in owr: https://github.com/EricssonResearch/openwebrtc/issues/186

**Comment by superdump** [comment]
This is now resolved, apart from #186. Closing.
